src/extension/ddappsec.c

```diff
@@ -208,8 +209,15 @@ static PHP_MSHUTDOWN_FUNCTION(ddappsec)
     return SUCCESS;
 }
 
+static void _dd_rinit_once(void) { dd_phpobj_load_env_values(); }
+
+static pthread_once_t dd_rinit_once_control = PTHREAD_ONCE_INIT;
+
 static PHP_RINIT_FUNCTION(ddappsec)
 {
+    // Things that should only run on the first RINIT
+    pthread_once(&dd_rinit_once_control, _dd_rinit_once);
```

Comment:
In ZTS, there are several threads serving requests, and each one has a copy of the ini entries. Therefore, this can't possibly work for ZTS. This is fine, because AFAICT this is only working around a pecularity of FPM, which doesn't use ZTS at all. But in that case:

1) all the code should be commented out is ZTS is enabled. Since there is only one thread for the PHP runtime, you also don't need to use pthreads.
2) for non-zts, it would make sense to check the if the SAPI is FPM and skip all this refreshing is not.

Reply:
Done!

Reply:
For the second part of 1., what I meant is that since you're not on ZTS, you can just a plain boolean as a guard:

```
// remove include to pthread.h
static boolean dd_rinit_once_done;
// ...
if (!dd_rinit_once_done) {
    _dd_rinit_once();
   dd_rinit_once_done = true;
}
```